app/script/entity/Conversation.js

```diff
@@ -461,24 +467,29 @@ z.entity.Conversation = class Conversation {
    * Checks for message duplicates.
    *
    * @private
-   * @param {z.entity.Message} messageEt - Message entity to be added to the conversation
+   * @param {z.entity.Message} messageEntity - Message entity to be added to the conversation
    * @returns {z.entity.Message|undefined} Message if it is not a duplicate
    */
-  _checkForDuplicate(messageEt) {
-    if (messageEt) {
-      for (const existingMessageEt of this.messages_unordered()) {
-        const duplicateMessageId = messageEt.id && existingMessageEt.id === messageEt.id;
-        const fromSameSender = existingMessageEt.from === messageEt.from;
-
-        if (duplicateMessageId && fromSameSender) {
-          const logData = {additionalMessage: messageEt, existingMessage: existingMessageEt};
-          this.logger.warn(`Filtered message '${messageEt.id}' as duplicate in view`, logData);
-          return undefined;
-        }
-      }
+  _checkForDuplicate(messageEntity) {
+    const existingMessageEntity = this._findDuplicate(messageEntity);
+    if (existingMessageEntity) {
+      const logData = {additionalMessage: messageEntity, existingMessage: existingMessageEntity};
+      this.logger.warn(`Filtered message '${messageEntity.id}' as duplicate in view`, logData);
+      return undefined;
     }
+    return messageEntity;
+  }
+
+  _findDuplicate(messageEntity) {
+    for (const existingMessageEntity of this.messages_unordered()) {
```

Comment:
Why not use ```this.messages_unordered().find()```?

Reply:
quick anwser: lazyness :D 

Good point :+1: